src/test/java/com/ea/restaurant/service/impl/Oauth2ServiceImplTest.java

```diff
@@ -109,10 +100,92 @@ public void whenWrongClientCredentialsLogin_ShouldReturnWrongCredentialsExceptio
         Assertions.assertThrows(
             RuntimeException.class,
             () -> {
-              var loginResponse = this.oauth2Service.loginClient("TEST02", "TEST-CLIENT-SECRET");
+              var loginResponse =
+                  this.oauth2Service.loginClient(
+                      Oauth2Fixture.TEST_WRONG_CLIENT_ID, Oauth2Fixture.TEST_CLIENT_SECRET);
             });
     var expectedWrongCredentialExceptionMessage = "Wrong Credentials";
 
     Assertions.assertEquals(exception.getMessage(), expectedWrongCredentialExceptionMessage);
   }
+
+  @Test
+  public void whenRefreshExpiredAccessToken_ShouldReturnNewAccessToken()
+      throws JOSEException, BadJOSEException, ParseException {
+
+    var mockedOauth2Util = MockOauth2Util.buildMockOauth2Util(appClient, appClientScope);
+    mockedOauth2Util
+        .when(
+            () ->
+                Oauth2Util.validateToken(
+                    Mockito.eq(AppAccessTokenFixture.EXPIRE_TOKEN),
+                    Mockito.eq(Oauth2Fixture.SECRET_KEY)))
+        .thenThrow(BadJWTException.class);
+
+    Mockito.when(
+            this.appRefreshTokenRepository.findByAccessTokenAndRefreshTokenAndClientId(
+                Mockito.eq(AppAccessTokenFixture.EXPIRE_TOKEN),
+                Mockito.eq(AppRefreshTokenFixture.TOKEN),
+                Mockito.eq(this.appClient.getClientId())))
+        .thenReturn(Optional.of(appRefreshToken));
+    Mockito.when(
+            this.appClientRepository.findByIdAndEntityStatus(
+                Mockito.eq(appRefreshToken.getAppClientId()), Mockito.eq(Status.ACTIVE)))
+        .thenReturn(Optional.of(appClient));
+    Mockito.when(
+            this.appClientScopeRepository.findByAppClientIdAndEntityStatus(
+                Mockito.eq(appClient.getId()), Mockito.eq(Status.ACTIVE)))
+        .thenReturn(Optional.of(appClientScope));
+    var refreshTokenResponse =
+        this.oauth2Service.refreshToken(
+            AppRefreshTokenFixture.TOKEN,
+            AppAccessTokenFixture.EXPIRE_TOKEN,
+            Oauth2Fixture.TEST_CLIENT_ID,
+            Oauth2Fixture.TEST_CLIENT_SECRET);
+
+    Assertions.assertNotEquals(
+        AppAccessTokenFixture.EXPIRE_TOKEN, refreshTokenResponse.getAccessToken());
+    Mockito.verify(appRefreshTokenRepository, Mockito.times(1))
+        .findByAccessTokenAndRefreshTokenAndClientId(
+            Mockito.eq(AppAccessTokenFixture.EXPIRE_TOKEN),
+            Mockito.eq(AppRefreshTokenFixture.TOKEN),
+            Mockito.eq(this.appClient.getClientId()));
+    Mockito.verify(appClientRepository, Mockito.times(1))
+        .findByIdAndEntityStatus(
+            Mockito.eq(appRefreshToken.getAppClientId()), Mockito.eq(Status.ACTIVE));
+    Mockito.verify(appClientScopeRepository, Mockito.times(1))
+        .findByAppClientIdAndEntityStatus(Mockito.eq(appClient.getId()), Mockito.eq(Status.ACTIVE));
+    mockedOauth2Util.close();
```

Comment:
You don't need this anymore I think

Reply:
As you need to deregistered static mock after each test it is necessary to implement `mockedStatic.close()`